src/meteodatalab/operators/time_operators.py

```diff
@@ -85,8 +93,10 @@ def delta(field: xr.DataArray, dtime: np.timedelta64) -> xr.DataArray:
     """
     nsteps = get_nsteps(field.valid_time, dtime)
     result = field - field.shift(time=nsteps)
-    result.attrs = field.attrs
-    return result
+    return xr.DataArray(
+        data=result,
+        attrs=metadata.override(field.message, typeOfStatisticalProcessing=4),
```

Comment:
Sorry I missed this on the first pass but I think this operator should set the length of time range